service-base/src/main/java/org/eclipse/hono/service/auth/device/UsernamePasswordAuthProvider.java

```diff
@@ -72,4 +78,25 @@ protected DeviceCredentials getCredentials(final JsonObject authInfo) {
         }
     }
 
+    @Override
+    protected Future<Device> validateCredentials(
+            final DeviceCredentials deviceCredentials,
+            final CredentialsObject credentialsOnRecord) {
+
+        final Context currentContext = Vertx.currentContext();
+        if (currentContext == null) {
```

Comment:
Since both branches (if and else) return at the end, it could be a perfect candidate to be written with the help of `Optional`. But more a matter of taste than a highly important improvement.